### ISSUE TITLE
Running Java examples on macOS

### ISSUE THREAD

**Issue by mcelvg** [mit-nlp/MITIE#125]
tl;dr -- on macOS 10.12, setting DYLD_LIBRARY_PATH doesn't add the mitielib directory to the `java.library.path`. A workaround is to set `java.library.path` from the command line OR copy the jnilib into a directory already on the `java.library.path` (both tested).

Cloning the latest version and following instructions in the [README.md](https://github.com/mit-nlp/MITIE/blob/master/README.md).

```
git clone https://github.com/mit-nlp/MITIE.git
cd MITIE
make MITIE-models

cd mitielib/java
mkdir build
cd build
cmake ..
cmake --build . --config Release --target install
```

Last few lines of output:
```
...
[100%] Built target javamitie
Install the project...
-- Install configuration: "Release"
-- Installing: ***/MITIE/mitielib/java/../libjavamitie.jnilib
-- Installing: ***/MITIE/mitielib/java/../javamitie.jar
```

and this is confirmed:
```
$ ls -L ../../*javamitie*
../../javamitie.jar       ../../libjavamitie.jnilib
```

but the NerExample fails:

```
$ cd ../../../examples/java

$ ./run_ner.sh 
loading NER model...
Native code library failed to load. 
java.lang.UnsatisfiedLinkError: no javamitie in java.library.path
```

Adding the following lines to `NerExample.java` will show the current `java.library.path`.

```
String property = System.getProperty("java.library.path");
for (String path : property.split(":"))
    System.err.println(path);
```

On this system, the default `java.library.path` includes:
```
~/Library/Java/Extensions
/Library/Java/Extensions
/Network/Library/Java/Extensions
/System/Library/Java/Extensions
/usr/lib/java
.
```

Copying the jnilib to any of these locations (including the current directory) works.

```
$ cp ../../mitielib/libjavamitie.jnilib .
$ ./run_ner.sh 
loading NER model...
Tags output by this NER model are: 
PERSON
LOCATION
ORGANIZATION
MISC
Number of entities found: 23
   Score: 0.566: ORGANIZATION:Pegasus Airlines 
   Score: 1.396: LOCATION:Istanbul 
   Score: 1.562: LOCATION:Sochi 
   Score: 1.293: LOCATION:Russia 
   Score: 1.136: LOCATION:Turkey 
   Score: 0.988: ORGANIZATION:Transportation Ministry 
   Score: 1.068: PERSON:Meredith Vieira 
   Score: 1.191: MISC:Olympics 
   Score: 0.880: PERSON:Bob Costas 
   Score: 1.262: PERSON:Vieira 
   Score: 0.704: MISC:Olympics 
   Score: 0.977: PERSON:Bob Costas 
   Score: 1.324: PERSON:Bob 
   Score: 0.736: ORGANIZATION:NBC Olympics 
   Score: 1.580: PERSON:Jim Bell 
   Score: 0.985: LOCATION:Sochi 
   Score: 1.491: PERSON:Josiah Franklin 
   Score: 1.513: PERSON:Benjamin Franklin 
   Score: 1.321: LOCATION:Boston 
   Score: 0.766: LOCATION:Philadelphia 
   Score: 0.978: PERSON:Benjamin Franklin 
   Score: 1.205: PERSON:Benjamin Franklin 
   Score: 1.143: LOCATION:Boston 
Relation detector type: people.person.place_of_birth
Benjamin Franklin 
BORN IN
Boston 

Benjamin Franklin 
BORN IN
Philadelphia 

Benjamin Franklin 
BORN IN
Boston 
```

Alternatively, it works if you change [run_ner.sh#L10](https://github.com/mit-nlp/MITIE/blob/master/examples/java/run_ner.sh#L10) to 
```
java -Djava.library.path="../../mitielib/" NerExample
```

This overrides the default `java.library.path`, which is probably fine for purposes of the examples.

This ticket may be related to [issue#79](https://github.com/mit-nlp/MITIE/issues/79).

I'm not certain why setting `DYLD_LIBRARY_PATH` doesn't work. Found a similar issue [here](https://github.com/oracle/node-oracledb/issues/231), which suggests it may be a security "feature" in recent versions of OSX. Have not tested to confirm.

====
cmake version 3.7.1
swig 3.0.6
jdk 1.8.0.45
clang 8.1.0.8020042
macOS 10.12.4

**Comment by davisking** [comment]
Thanks for the detailed bug report :)

Do you want to submit the change as a PR (so you get github points for it :) ), or should I make the change?

**Comment by mcelvg** [comment]
Sure, I'll submit a PR. Thanks.
